code/script.js

```diff
@@ -218,108 +256,233 @@ const generateBoard = () => {
+// const guess = (personToConfirm) => {
+//   let confirmation = confirm("You want to make a guess?");
+//   if (confirmation) {
+//     checkMyGuess(personToConfirm);
+//   }
+// };
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
-}
-
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// Connecting a customized confirmation popup https://getbootstrap.com/docs/4.0/components/modal/
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
+  $("#myModal").modal("show"); /* Thats a bootstrap modal component - called with use of jQuery*/
+  selectedPerson = personToConfirm;
+};
 
-// If you confirm, this function is invoked
+// If user confirms that he/she wants to make a guess, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  if (personToCheck === secret.name) {
+    showWinScreen();
+    /* playes audio for a winner on the Desktop version*/
+    audio.play();
+    /* to control the audio volume*/
+    audio.volume = 0.2;
+  } else {
+    showLoseScreen();
+  }
+};
+
+// This function is drawing the winning screen provides an info on number of attempts being made, showing
+// the picture of correct character
+const showWinScreen = () => {
+  winOrLoseText.innerText = `You Won in ${attempts} attempts!`;
+  winOrLoseSection.style.display = "flex";
+  winCard.innerHTML = `
+  <div class="card">
+    <p>${secret.name}</p>
+    <img src=${secret.img}>
+  </div>`;
+  board.style.display = "none";
+  alertWrapper.remove();
+};
+
+// This function is drawing the losing screen
+const showLoseScreen = () => {
+  winOrLoseText.innerText = "Sorry, Wrong Guess!";
+  winOrLoseSection.style.display = "flex";
+  board.style.display = "none";
+  alertWrapper.remove();
+};
+
+// timer creation: https://stackoverflow.com/questions/5517597/plain-count-up-timer-in-javascript
+setInterval(setTime, 1000);
+
+// counting function
+function setTime() {
+  ++totalSeconds;
+  secondsLabel.innerHTML = pad(totalSeconds % 60);
+  minutesLabel.innerHTML = pad(parseInt(totalSeconds / 60));
 }
 
-// Invokes the start function when website is loaded
-start()
+// timer rendering
+function pad(val) {
+  let valString = val + "";
+  if (valString.length < 2) {
+    return "0" + valString;
+  } else {
+    return valString;
+  }
+}
 
```

Comment:
Great job learning about and adding yet another feature. I think you learned a lot of different things that js can do this week!